api/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaHttpClient.java

```diff
@@ -61,21 +78,41 @@ final class JavaHttpClient implements org.projectnessie.client.http.HttpClient {
     }
 
     client = clientBuilder.build();
+    writerPool = new ForkJoinPool(Math.max(8, ForkJoinPool.getCommonPoolParallelism()));
   }
 
   @Override
   public HttpRequest newRequest(URI baseUri) {
-    return new JavaRequest(this.config, baseUri, client::send);
+    return new JavaRequest(this.config, baseUri, client::send, writerPool);
   }
 
   @Override
   public URI getBaseUri() {
     return config.getBaseUri();
   }
 
+  @SuppressWarnings({"DataFlowIssue", "ResultOfMethodCallIgnored"})
   @Override
   public void close() {
-    client = null;
-    config.close();
+    try {
+      if (client instanceof AutoCloseable) {
+        ((AutoCloseable) client).close();
+      }
+    } catch (Exception e) {
+      // ignore
+    } finally {
+      client = null;
+    }
+    try {
+      config.close();
+    } finally {
+      writerPool.shutdown();
+      try {
+        writerPool.awaitTermination(10, TimeUnit.SECONDS);
+      } catch (InterruptedException e) {
+        // populate
```

Comment:
populate?

Reply:
yea - to set the flag again in the line below

Reply:
Did you mean `propagate`?